backtrace-library/src/main/java/backtraceio/library/services/BacktraceNativeCrashResponseListener.java

```diff
@@ -0,0 +1,127 @@
+package backtraceio.library.services;
+
+import android.util.Log;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.ArrayList;
+import java.util.List;
+
+import backtraceio.library.BacktraceClient;
+import backtraceio.library.base.BacktraceBase;
+import backtraceio.library.events.OnServerResponseEventListener;
+import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.models.BacktraceResult;
+import backtraceio.library.models.json.BacktraceReport;
+import backtraceio.library.models.types.BacktraceResultStatus;
+
+public class BacktraceNativeCrashResponseListener
+        implements OnServerResponseEventListener {
+    private static transient final String LOG_TAG = BacktraceNativeCrashResponseListener.
+            class.getSimpleName();
+
+    private static BacktraceNativeCrashResponseListener instance;
+    private static BacktraceClient client;
+    private static String databasePath;
+
+    private BacktraceNativeCrashResponseListener() {}
+
+    @Override
+    public void onEvent(BacktraceResult backtraceResult) {
+        if (backtraceResult.status == BacktraceResultStatus.Ok) {
+            // Get the minidump out
+            BacktraceReport report = backtraceResult.getBacktraceReport();
+            if (report != null) {
+                List<String> attachments = report.attachmentPaths;
+                for (String attachment : attachments) {
+                    if (attachment.endsWith(".dmp")) {
+                        moveToCompleted(attachment);
+                        break;
```

Comment:
why only first one?

Reply:
There should only be one minidump file, so once it's found we don't need to continue the iteration.  Since we're looking through the crashpad database directories (/new and /pending) for minidumps each of those minidumps will get processed separately so there isn't a case of having more than one.